### PR TITLE
added style to login page

### PR DESCRIPTION
**Pull request by manema**
### Main Title: Login Page

**Trello card:** 
https://trello.com/c/6NvE0E4O/17-2-as-a-user-i-should-be-able-to-sign-in-using-my-email-and-password

**Description:**
As a user I should be able to sign in using my email and password

**Changes:**
- Added style to login screen.
- Some components have been refactored in order to use these in the future.

login_screen com-video-to-gif (1)

### CODE REVIEW

File: src/components/common/WhiteButton/styles.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { StyleSheet } from 'react-native';
+
+export const style = (alignSelf, borderBottomColor, borderBottomWidth, marginVertical,
+  width) =>
+  StyleSheet.create({
```
Comment (agustito37) [inline src/components/common/WhiteButton/styles.js]: Alo! Don't use create for styles that depend on parameters.
```suggestion
  ({
```

File: src/containers/LoginScreen/index.js [hunk 2]
```diff
@@ -1,25 +1,59 @@
...
 import LoginForm from 'components/user/LoginForm';
 import { login } from 'actions/userActions';
 import translate from 'utils/i18n';
+import Separator from 'components/common/Separator';
+import { BLACK, WHITE } from 'constants/styleConstants';
+import { isIos } from 'constants/appConstants';
+import MainHeader from 'components/common/MainHeader';
+import background from 'assets/background.png';
+import Button from 'components/common/Button';
+import WhiteButton from 'components//common/WhiteButton';
```
Comment (agustito37) [inline src/containers/LoginScreen/index.js]:
```suggestion
import WhiteButton from 'components/common/WhiteButton';
```

File: src/containers/LoginScreen/index.js [hunk 3]
```diff
@@ -1,25 +1,59 @@
...
-    </Text>
-    <LoginForm onSubmit={user => login(user.toJS())} />
-    <Button
-      title={translate('SIGN_UP.title')} onPress={() => navigator.push({
-        screen: 'reactnativebase.SignUpScreen'
-      })}
-    />
-  </View>
+  <ScrollView>
+    <ImageBackground resizeMode="cover" source={background} style={{ flex: 1 }}>
```
Comment (agustito37) [inline src/containers/LoginScreen/index.js]: Can you move this inline style to the styles file?

File: src/containers/LoginScreen/index.js [hunk 4]
```diff
@@ -1,25 +1,59 @@
...
+      <View style={styles.container}>
+        <MainHeader text={translate('MAIN_SCREEN.title')} />
+        <LoginForm onSubmit={user => login(user.toJS())} />
+        <WhiteButton
+          title={translate('SIGN_UP.forgottenPassword')}
+          onPress={() => {}}
+          marginTop={3}
```
Comment (agustito37) [inline src/containers/LoginScreen/index.js]: Maybe you can start importing these values from the styles files; it's a good practice, given that they are style units.

Comment (missx) [reply src/containers/LoginScreen/index.js]: Yeah, I agree, is it necessary to have all these styles added dynamically to the component? Will you have a lot of different WhiteButton components styled differently?

File: src/components/user/SignUpForm/index.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ import Input from 'components/common/Input';
 import Custompicker from 'components/common/CustomPicker';
 import translate from 'utils/i18n';
 import { GENDER } from 'constants/appConstants';
-import { BLACK, WHITE, FONT_ERROR } from 'constants/styleConstants';
+import { FONT_ERROR } from 'constants/styleConstants';
 import Button from '../../common/Button';
```
Comment (jdalto) [inline src/components/user/SignUpForm/index.js]: You can use shortcut path here too

File: src/containers/SignUpScreen/index.js [hunk 6]
```diff
@@ -1,32 +1,28 @@
...
+import WhiteButton from 'components/common/WhiteButton';
+import Separator from 'components/common/Separator';
+import MainHeader from 'components/common/MainHeader';
+import background from 'assets/background.png';
+import { isIos } from 'constants/appConstants';
 import styles from './styles';
 
 const SignUpScreen = ({ signUp, navigator }) =>
   <ScrollView>
     <ImageBackground resizeMode="cover" source={background} style={{ flex: 1 }}>
```
Comment (jdalto) [inline src/containers/SignUpScreen/index.js]: Remove this inline style

File: src/components/common/MainHeader/index.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (peter7z) [inline src/components/common/MainHeader/index.js]:
```suggestion
import { string } from 'prop-types';
```

And then remove line 12

File: src/components/user/LoginForm/index.js [hunk 8]
```diff
@@ -1,15 +1,16 @@
 import React from 'react';
 import PropTypes from 'prop-types';
```
Comment (peter7z) [inline src/components/user/LoginForm/index.js]: Same here. You can directly deconstruct the module in the import line.
Check for this in other files too.

**Pull request closed without merging**